src/main/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfiguration.java

```diff
@@ -218,19 +217,33 @@ public OpenTelemetryConfiguration toOpenTelemetryConfiguration() {
             configurationProperties);
     }
 
+    /**
+     * Register reconfigurable {@link io.opentelemetry.api.OpenTelemetry}
+     * on {@link  io.opentelemetry.api.GlobalOpenTelemetry}
+     * and {@link io.jenkins.plugins.opentelemetry.opentelemetry.ReconfigurableEventLoggerProvider}
+     * on {@link io.opentelemetry.api.incubator.events.GlobalEventLoggerProvider}
+     * as early as possible in Jenkins lifecycle so any plugin invoking those Global setters will have the
+     * reconfigurable instance .
+     */
+    @Initializer(after = InitMilestone.EXTENSIONS_AUGMENTED, before = InitMilestone.SYSTEM_CONFIG_LOADED)
```

Comment:
before system config loaded? how will the configuration be available for this plugin? and how will someone configure this plugin with JCasC if its so early?

jcasc is after system config loaded:
https://github.com/jenkinsci/configuration-as-code-plugin/blob/3b0e6de4ab83e8024e6cac635728c28b68ac663a/plugin/src/main/java/io/jenkins/plugins/casc/ConfigurationAsCode.java#L337

Reply:
Correct: There reasons is:
1. we want to be sure that any plugin that is using OTel will use our implementation of `OpenTelemetry`, even when getting this OpenTelemetry instance through `GlobalOpenTelemetry.get()`
2. we don't know how to ensure the OpenTelemetry plugin gets configured & initialized before all other plugins (e.g. the initialization code [`DatabaseSchemaLoader#migrateSchema()`](https://github.com/jenkinsci/junit-sql-storage-plugin/blob/76cefd5f4cf6117553fa6f219442a9cbba64b458/src/main/java/io/jenkins/plugins/junit/storage/database/DatabaseSchemaLoader.java#L24-L26) of the JUnit SQL Storage Plugin is invoked by Jenkins before the Jenkins OTel Plugin initialization code [`JenkinsOpenTelemetryPluginConfiguration#initializeOpenTelemetry()`](https://github.com/jenkinsci/opentelemetry-plugin/blob/3aa8270a1fd83c8b53014ac66c448bb0b33108fd/src/main/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfiguration.java#L242C17-L242C40) despite the dependency.

The solution we identified is to:
1. Register a Reconfigurable OpenTelemetry instance in `GlobalOpenTelemetry` as early as possible in Jenkin's lifecycle with a NoOp implementation (see `initializeOpenTelemetryAfterExtensionsAugmented()`).
2. When the Jenkins Otel Plugin config is ready, we reconfigure the OpenTelmetry instance (see `initializeOpenTelemetry()`)

I think we can cleanup a bit this code but it works and it was quite time consuming to get there :-) 

Dos it make sense?

https://github.com/jenkinsci/junit-sql-storage-plugin/blob/76cefd5f4cf6117553fa6f219442a9cbba64b458/src/main/java/io/jenkins/plugins/junit/storage/database/DatabaseSchemaLoader.java#L24-L26

Reply:
ah I see bit confusing but it'll work I think